### PR TITLE
Assume role and other recipe cleanup

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**
@ranchodeluxe tagging as an FYI since I will just merge this into mursst-kerchunk after testing. This PR:

- removes HTTPS as a way to access the data. I discussed this with Sean and Raphael and they agreed we should remove it since it's not really recommended to access the data that way on this scale.
- removes unused transforms and functions
- adds configurable `auth_mode` to be `iamrole`, `assumerole` or `edl` credentials endpoint.
- Uses a ParDo / DoFn to setup credentials. This is a workaround until we have a solution integrated into pangeo-forge-recipes and pangeo-forge-runner that can handle setting credentials and resetting them when they expire.

Note: I just added the the option `assumerole` which is intended for what we were previously using `iamrole` for -- that is, assuming whatever role is set as as the `AWS_ROLE_ARN` -- because it makes sense to instead use `iamrole` for when we want to use the already set environment variable `AWS_ROLE_ARN` for data access, which is what we want to do in the case we are using the VEDA JupyterHub.

### CODE REVIEW

File: recipes/mursst/recipe.py [hunk 1]
```diff
@@ -34,141 +32,75 @@
...
-            # see Kerchunk docs for dropping a variable example
-            translated['refs'] = {
-                k: v
-                for k, v in translated['refs'].items()
-                if '/' not in k or k.split('/')[0] in keep
+            import boto3
+            client = boto3.client('sts')
+            creds = client.assume_role(
+                RoleArn=aws_role_arn,
+                RoleSessionName='mursst-pangeo-forge',
```
Comment (ranchodeluxe) [inline recipes/mursst/recipe.py:66]: @abarciauskas-bgse: Should we add a `DurationSeconds=43200` arg here too (which is the max I think). I think the default `SessionToken` expiry here is 3600 seconds (1 hours) and might put us back into the same situation that we had with EDL

Comment (abarciauskas-bgse) [reply recipes/mursst/recipe.py:66]: @ranchodeluxe I added the DurationSeconds to the max (43200 seconds, 12 hours) as you suggested. It sounds like you already updated this on the role, thank you!

Do you think this is good to merge now. It's just into the mursst-kerchunk branch.

Comment (abarciauskas-bgse) [reply recipes/mursst/recipe.py:66]: I just logged into SMCE VEDA account and the `veda-data-reader-dev` Maximum session duration was only 1 hour so I increased it to 12 hours.

I also opened a PR to use this role in veda-pforge-job-runner: https://github.com/NASA-IMPACT/veda-pforge-job-runner/pull/48

**Comment by abarciauskas-bgse** [comment]
Finally got this recipe to run all days from 2002-06-01 to 2023-02-23 (when the chunking changed).

# Details

## Command used:
```
AUTH_MODE=iamrole pangeo-forge-runner bake \
  --repo="pangeo_forge_staging" \
  --ref="ab/murssst-kerchunk-assume-role" \
  --feedstock-subdir="recipes/mursst" \
  --Bake.recipe_id=MUR-JPL-L4-GLOB-v4.1 \
  --Bake.job_name=local-mursst-kerchunk \
  -f multi-local-runner-config.py
```

## Infrastructure:
* VEDA JupyterHub with 120GB RAM and 15.7 CPUs
* Used 30GB memory and >200% CPU
Screenshot 2024-03-14 at 2 26 17 PM
* Took about an hour

## Config

```
# multi-local-runner-config.py
c.Bake.bakery_class = "pangeo_forge_runner.bakery.local.LocalDirectBakery"
c.LocalDirectBakery.num_workers = 10

c.MetadataCacheStorage.fsspec_class = "fsspec.implementations.local.LocalFileSystem"
# Metadata cache should be per `{{job_name}}`, as kwargs changing can change metadata
c.MetadataCacheStorage.root_path = "./metadata"

c.TargetStorage.fsspec_class = "fsspec.implementations.local.LocalFileSystem"
c.TargetStorage.root_path = "./multi-runner-output"
```

## Validation

This notebook provides some anecdotal validation https://gist.github.com/abarciauskas-bgse/00d3db25e00ff0da33696597dc94ed3d

@ranchodeluxe Do think we can merge this and update the credential generation later on?

**Review by ranchodeluxe — APPROVED** [review]
👍

**Pull request merged**